internal/db/sqltest/tests/iam/grant_scope_validation.sql

```diff
@@ -0,0 +1,139 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+select plan(13);
+
+-- Remove any existing roles, which should also remove existing grant scopes
+delete from iam_role;
+
+-- Set up a series of scopes to test against
+insert into iam_scope (type, public_id, parent_id) values
+    ('org', 'o_1111111111', 'global'),
+    ('project', 'p_111111111a', 'o_1111111111'),
+    ('project', 'p_111111111b', 'o_1111111111'),
+    ('org', 'o_2222222222', 'global'),
+    ('project', 'p_222222222a', 'o_2222222222'),
+    ('project', 'p_222222222b', 'o_2222222222');
```

Comment:
nit: Here and below, if you can align the columns and values. We do this elsewhere as it helps with readability of the insert.

```suggestion
insert into iam_scope
    (type,      public_id,      parent_id)
values
    ('org',     'o_1111111111', 'global'),
    ('project', 'p_111111111a', 'o_1111111111'),
    ('project', 'p_111111111b', 'o_1111111111'),
    ('org',     'o_2222222222', 'global'),
    ('project', 'p_222222222a', 'o_2222222222'),
    ('project', 'p_222222222b', 'o_2222222222');
```